### PR TITLE
Plug memory leaks

### PR DESCRIPTION
**Pull request by qpernil**

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -1766,6 +1766,16 @@ int validate_and_call(yubihsm_context *ctx, CommandList l, const char *line) {
   return 0;
 }
 
+static void free_configured_connectors(yubihsm_context *ctx) {
+  if (ctx->connector_list) {
```
Comment (a-dma) [inline src/main.c:1770]: I would change to `if (ctx && ctx->connector_list)`

File: src/main.c [hunk 2]
```diff
@@ -1790,6 +1796,17 @@ static int parse_configured_connectors(yubihsm_context *ctx, char **connectors,
 }
 
 #ifdef USE_ASYMMETRIC_AUTH
+
+static void free_configured_pubkeys(yubihsm_context *ctx) {
+  if (ctx->device_pubkey_list) {
```
Comment (a-dma) [inline src/main.c:1801]: similarly `if (ctx && ctx->device_pubkey_list)`

File: src/main.c (hunk 1 above)
Comment (qpernil) [reply src/main.c:1770]: Yeah, except other similar functions don't do it...

File: src/main.c (hunk 2 above)
Comment (qpernil) [reply src/main.c:1801]: Personally I would then also do it in other similar functions. And I'm also reluctant to maybe inadvertently hide bugs.

Comment (qpernil) [reply src/main.c:1801]: In this case, with the usage we have, we know it's not needed as well

Comment (a-dma) [reply src/main.c:1801]: given that it's an internal function it's OK not to check if we can guarantee pre-conditions. The main risk is if something changes in the future it becomes hard to keep track of this kind of things.

Comment (qpernil) [reply src/main.c:1801]: Yes. But just keeping with the style of similar code around it.

**Review by a-dma — APPROVED** [review]

**Pull request merged**